src/test/java/jimmy/mcgymmy/model/food/ProteinTest.java

```diff
@@ -0,0 +1,38 @@
+package jimmy.mcgymmy.model.food;
+
+import static jimmy.mcgymmy.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class ProteinTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NumberFormatException.class, () -> new Protein(null));
+    }
+
+    @Test
+    public void constructor_invalidProtein_throwsIllegalArgumentException() {
+        String invalidProtein = "";
+        assertThrows(IllegalArgumentException.class, () -> new Protein(invalidProtein));
+    }
+
+    @Test
+    public void isValidProtein() {
```

Comment:
Should use the module convention, i.e. `protein_isValid_correct` or something

Same for the methods in the other test classes